lib/queue.js

```diff
@@ -423,7 +430,10 @@ Queue.prototype.run = function(concurrency){
   var promises = [];
   var _this = this;
 
-  return this.processStalledJobs().then(function(){
+  // In case of connection loss, running `processStalledJobs` will repick jobs here.
+  var start = this.opts.processStalledJobs ? this.processStalledJobs() : Promise.resolve();
```

Comment:
The expectations of Bull is to process them immediately on reconnection. Not calling it to begin with result in many failures across tests.